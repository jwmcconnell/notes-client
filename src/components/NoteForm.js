import React from 'react';
import PropTypes from 'prop-types';

const NoteForm = () => {
  return (
    <h2>Note Form</h2>
  );
};

NoteForm.propTypes = {

};

export default NoteForm;
