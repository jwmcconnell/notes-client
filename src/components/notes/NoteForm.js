import React from 'react';
import PropTypes from 'prop-types';

const NoteForm = ({ 
  title,
  handleTitleChange,
  body,
  handleBodyChange,
  handleSubmit,
}) => {
  return (
    <form onSubmit={handleSubmit}>
      <label>Title:
        <input type="text" placeholder="Title" value={title} onChange={handleTitleChange} />
      </label>
      <label>Body:
        <textarea 
          name="text" 
          placeholder="Body"
          value={body}
          id="text" 
          cols="30" 
          rows="10" 
          onChange={handleBodyChange}
        ></textarea>
      </label>
      <button>Submit</button>
    </form>
  );
};

NoteForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleTitleChange: PropTypes.func.isRequired,
  handleBodyChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default NoteForm;
