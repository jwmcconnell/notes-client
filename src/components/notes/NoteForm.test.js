import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('NoteForm component', () => {
  it('renders NoteForm', () => {
    const wrapper = shallow(<NoteForm 
      title="hi" 
      body="there" 
      handleTitleChange={() => { }}
      handleBodyChange={() => { }}
      handleSubmit={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
