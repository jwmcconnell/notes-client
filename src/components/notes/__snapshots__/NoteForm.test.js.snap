// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm component renders NoteForm 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Title:
    <input
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value="hi"
    />
  </label>
  <label>
    Body:
    <textarea
      cols="30"
      id="text"
      name="text"
      onChange={[Function]}
      placeholder="Body"
      rows="10"
      value="there"
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
